Search Keyboard Navigation

Hackin7 opened the pull request:
![image](https://user-images.githubusercontent.com/18213722/182663092-b204c237-cd0f-4664-9fdf-6684644b1d67.png)

wei2912 commented:
1. TODO: Search bar should refocus when pressing up key from first search result.
2. After this PR is merged: Implement first search result being selected by default, so that people can type a search query and hit the enter bar once they get a satisfactory result.

Hackin7 commented:
Added Features and fixed some bugs like 
1. The case if the user is not focused on the modalInput, makes the UI work better
2. move up to select the input field
3. Type anywhere to fill up the input, Enter to select the option
4. Auto select the first option (with debounce, a bit janky)

wei2912 commented:
1. Search relevance to be tackled in future PR
2. When focused text field, down arrow key should go to end of text box.

soloplxya commented on assets/js/main.js:
```diff
@@ -296,23 +296,39 @@ function elasticSearch(query, callback) {
+            clearTimeout(focusOnFirst);
+            focusOnFirst = setTimeout(function () {
+                if (searchListingLength == 0) return;
+                if (searchSelectionId >= 0) return;
```
Should it be `searchSelectionId > 0` since we are trying to select the first element?

Hackin7 replied:
the index of the first element is 0, hence should be >=0. searchSelectionId = -1 in this case represents the search box
As for the AdvisorySG entry, I think its an ElasticSearch (Backend Issue), so maybe it could be spun off into another issue

soloplxya reviewed (approved):
![image](https://user-images.githubusercontent.com/69782911/194124512-13d60490-fb1f-4d02-9d82-5ad78d6754df.png)

Looks good overall! But I think the search result if possible (can be improved) such that we won't see `AdvisorySG` appear when we search `Conversations with...`.
